subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaPlugin.java

```diff
@@ -50,16 +51,15 @@ public void apply(Project project) {
         project.getPluginManager().apply(JavaPlugin.class);
 
         final SourceSet main = project.getExtensions().getByType(JavaPluginExtension.class).getSourceSets().getByName("main");
-
-        configureScaladoc(project, main);
-
         final Configuration incrementalAnalysisElements = project.getConfigurations().getByName("incrementalScalaAnalysisElements");
         String compileTaskName = main.getCompileTaskName("scala");
         final TaskProvider<AbstractScalaCompile> compileScala = project.getTasks().withType(AbstractScalaCompile.class).named(compileTaskName);
         final Provider<RegularFile> compileScalaMapping = project.getLayout().getBuildDirectory().file("tmp/scala/compilerAnalysis/" + compileTaskName + ".mapping");
         compileScala.configure(task -> task.getAnalysisMappingFile().set(compileScalaMapping));
         incrementalAnalysisElements.getOutgoing().artifact(
             compileScalaMapping, configurablePublishArtifact -> configurablePublishArtifact.builtBy(compileScala));
+            
+        configureScaladoc(project, main, compileScala);
```

Comment:
This is what fails the build - there is no third parameter in `configureScaladoc`

Reply:
Yeah, my bad, sorry